phpBB/phpbb/search/backend/fulltext_native.php

```diff
@@ -83,49 +88,48 @@ class fulltext_native extends \phpbb\search\base
 
 	/**
 	 * Config object
-	 * @var \phpbb\config\config
+	 * @var config
 	 */
 	protected $config;
 
 	/**
 	 * Database connection
-	 * @var \phpbb\db\driver\driver_interface
+	 * @var driver_interface
 	 */
 	protected $db;
 
 	/**
 	 * phpBB event dispatcher object
-	 * @var \phpbb\event\dispatcher_interface
+	 * @var dispatcher_interface
 	 */
 	protected $phpbb_dispatcher;
 
 	/**
 	 * User object
-	 * @var \phpbb\user
+	 * @var user
 	 */
 	protected $user;
 
 	/**
-	* Initialises the fulltext_native search backend with min/max word length
-	*
-	* @param	boolean|string	&$error	is passed by reference and should either be set to false on success or an error message on failure
-	* @param	string	$phpbb_root_path	phpBB root path
-	* @param	string	$phpEx	PHP file extension
-	* @param	\phpbb\auth\auth	$auth	Auth object
-	* @param	\phpbb\config\config	$config	Config object
-	* @param	\phpbb\db\driver\driver_interface	$db	Database object
-	* @param	\phpbb\user	$user	User object
-	* @param	\phpbb\event\dispatcher_interface	$phpbb_dispatcher	Event dispatcher object
-	*/
-	public function __construct(&$error, $phpbb_root_path, $phpEx, $auth, $config, $db, $user, $phpbb_dispatcher)
+	 * Initialises the fulltext_native search backend with min/max word length
+	 *
+	 * @param config $config Config object
+	 * @param driver_interface $db Database object
+	 * @param dispatcher_interface $phpbb_dispatcher Event dispatcher object
+	 * @param user $user User object
+	 * @param string $phpbb_root_path phpBB root path
+	 * @param string $phpEx PHP file extension
+	 */
+	public function __construct($config, $db, $phpbb_dispatcher, $user, $phpbb_root_path, $phpEx)
```

Comment:
Any reason why you're not specifying the parameter type here?

Reply:
No, I mostly moved code here, I didn't rewrite the classes